build/cmake/CMakeLists.txt

```diff
@@ -115,9 +115,20 @@ endif ()
 # External dependencies
 #-----------------------------------------------------------------------------
 if (ZSTD_MULTITHREAD_SUPPORT AND UNIX)
-    set(THREADS_PREFER_PTHREAD_FLAG ON)
-    find_package(Threads REQUIRED)
-    if(CMAKE_USE_PTHREADS_INIT)
+    if (CMAKE_SYSTEM_NAME MATCHES "HP-UX")
+        find_package(Threads)
+        if (NOT Threads_FOUND)
```

Comment:
If I understand correctly, this patch covers an exception scenario, where :
- OS is `HP-UX`
- `find_package(Threads)` _may_ be successful (?), but it may also fail
- when it fails, the script enters this branch
- The actions in this branch essentially force using `pthread`, even though it was not found by `find_package(Threads)`, by initializing the relevant flags

Is that a correct interpretation ?

Reply:
Yes.

* On HP-UX 11.31 IA64  , CMake 3.8.2 with GCC 4.7.1 `find_package(Threads)` successfully.
* On HP-UX 11.11 PA-RISC,   CMake 3.8.2 with GCC 4.7.1 `find_package(Threads)` error.

However, it is hard to detect whether the architecture is PA-RISC by CMake:

```bash
 /bin/uname -a # HP-UX 11.31 IA64
HP-UX hprx2660 B.11.31 U ia64 3240820735 unlimited-user license


/bin/uname -a # HP-UX 11.11 PA-RISC

HP-UX hpux11iv B.11.11 U 9000/800 360203311 unlimited-user license
```

I'm not sure whether `9000/800` denotes PA-RISC.

Reply:
OK, thanks, that's clearer.

It's probably worth explaining in comments,
because this distinction between "HP-UX" variants `PA-RISC` vs `IA64` and their consequences is not obvious from the code.